### PR TITLE
Update pycparser and PyYAML versions

### PR DESCRIPTION
**Pull request by soohoonchoi**
This PR addresses issues encountered during a white box build for Chapel on Shasta. Specifically, the current version build for Shasta was encountering "ModuleNotFoundError: No module named 'six'".  Updating the pycparser and PyYAML have resolved that issue.

### CODE REVIEW

File: third-party/chpl-venv/c2chapel-requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 # tools/c2chapel/Makefile should match so fakeHeaders download matches
```
Comment (mppf) [inline third-party/chpl-venv/c2chapel-requirements.txt:1]: as the comment says you should update the version in tools/c2chapel/Makefile

Comment (soohoonchoi) [reply third-party/chpl-venv/c2chapel-requirements.txt:1]: thank you for the review.  I've updated the version.

**Review by lydia-duncan — APPROVED** [review]
Thanks!

**Comment by mppf** [comment]
I've run this through testing and I see one failure:

```
[Error matching program output for c2chapel/c2chapel-tester]
```

I'm not sure what's going on there but it looks like there are failures in 2 tests.

**Comment by bradcray** [comment]
I came looking for this PR because we started talking about a 1.24.1 release today, which made me curious what had ended up happening here.

I took a look to see if I could reproduce the c2chapel test failures that Michael noted, and in my first attempt with the branch, found four failures falling into two varieties on my Mac.  However, since the number differed from Michael's report, I then tried it on linux64 with the branch and got zero failures.  Then, I went back to Mac, checked out the master branch, tried to clean up, and got only two failures of the first variety below.

At this point, I found myself wondering whether everything is fine and whether we just don't test this on very many system configurations.  But then I went back to the branch on linux, tried to clean up and rebuild, and started getting two failures of the second variety below.  Then I went back to Mac, did a similar clean up and rebuild, and got all four failures below.  So my guess is that the first flavor of problem is Mac-specific (or "my Mac's venv"-specific?), but that the second class of failure is a real difference (but maybe not an important one?).  And it seems that we only test c2chapel in one configuration (chapcs09).

The first variety of failure that I saw on the Mac, and only the Mac was as follows:

```
<                 file [cppFlags ...]
---
>                 file [cppFlags [cppFlags ...]]
```
This actually looks as if it's only more correct to me than it was before.  At least, I'm not seeing any good reason for the nested `cppFlags` optional argument in the previous output.  So my thought was to just accept the .good and take the change.  But again, this is only happening on my Mac, so I suppose we should leave it be since we're not testing c2chapel on Mac

The second failure mode was an output difference as follows:

```
< // ==== c2chapel typedefs ====
<
< // Opaque struct?
< extern record MirBlob {};
<
< // Opaque struct?
< extern record MirBufferStream {};
<
< // Opaque struct?
< extern record MirConnection {};
<
< // Opaque struct?
< extern record MirDisplayConfig {};
<
< extern type MirEGLNativeDisplayType = c_void_ptr;
<
< extern type MirEGLNativeWindowType = c_void_ptr;
<
< // Opaque struct?
< extern record MirPersistentId {};
<
< // Opaque struct?
< extern record MirPromptSession {};
<
< // Opaque struct?
< extern record MirScreencast {};
<
< // Opaque struct?
< extern record MirSurface {};
<
< // Opaque struct?
< extern record MirSurfaceSpec {};
<
< // Opaque struct?
< extern record xcb_connection_t {};
<
< extern type xcb_visualid_t = uint(32);
<
< extern type xcb_window_t = uint(32);
<
```

These only appear in the two tests that `#include <std*.h>`, and my guess is that the new version of pycparser is somehow causing the new `Mir*` data structures to be included and generated (or maybe our test script doesn't screen them out as well as it does other things?  Though I'm not seeing any code that seems to be doing any filtering...).  Googling the names, they seem to be from this package:  https://mir-server.io/doc/group__mir__toolkit.html

But I don't see the change as being deep or meaningful.  So I think the main question is whether to put them into the expected output for these tests or to put more work into the `tester.sh` script to filter them out.  (and maybe, optionally, try to understand what change causes them to get included).

Tagging @ben-albrecht here who seems to have done the original PR review for BHarsh's addition of the c2chapel tests in case he has any insights or inputs.

**Comment by leekillough** [comment]
Cherry-pick [this fix](https://github.com/leekillough/chapel/commit/30925e027f0649538749d56bd4eb7a4d851abd86) to see if it helps with `c2chapel` tests.

**Comment by bradcray** [comment]
> Cherry-pick this fix to see if it helps with c2chapel tests.

With a quick check, it doesn't seem to (nor does it seem related to the failure modes).

**Comment by bradcray** [comment]
I've got a lead on what's going on here, though not a solution yet.  Following up on this comment:

> These only appear in the two tests that #include <std*.h>, and my guess is that the new version of pycparser is somehow causing the new Mir* data structures to be included and generated (or maybe our test script doesn't screen them out as well as it does other things? Though I'm not seeing any code that seems to be doing any filtering...).

I'm learning that the thing which causes `c2chapel` to not emit declarations for every symbol in `stdint.h`, `stdlib.h`, `stdio.h`, etc. is the `tools/c2chapel/install/fakeHeaders/` directory, which is populated with little stub `#include` files for each of the standard header files.  The contents of this directory are themselves created by running `pycparser` which is one of the things that had a version change in this PR.

So then, there are two possibilities:  Either the header file that defines these `Mir*` data structures is not among these fake headers such that we're getting the real header; or something in the fake headers is defining those `Mir*` data structures.  

Doing a `find . -name "*.h" -exec grep --with-filename "Mir" {} \;` in the fakeHeaders directory, it seems to be the latter, as `_fake_typedefs.h` defines these `Mir*` data structures on this branch (but presumably doesn't on master).

So then the questions become:

* What is the role of `_fake_typedefs.h`?  Should it contain these `Mir*` typedefs or not?
  * if so, then we probably just need to update the .good files to reflect it
  * if not, then we should probably update the process that builds it to remove them or filter them out or whatever...

**Comment by bradcray** [comment]
As an update, Lee has determined that (a) pycparser generates the `fakeHeaders` contents, and (b) that it was changed in [this commit](https://github.com/eliben/pycparser/commit/c7b929cab09fc4332e13f3ecaa626e2afb082a9d) to include Mir, so that part makes sense.  The main remaining question is why the c2chapel output puts it in the section labeled

```
// ==== c2chapel typedefs ====
```

rather than the section labeled:

```
// c2chapel thinks these typedefs are from the fake headers:
```

He's looking into this now.

**Comment by ronawho** [comment]
I think I have a fix for this. c2chapel tries to ignore things that are in fake_headers, but it hardcodes patterns to ignore, so we just have to change our filters to adjust for some new patterns:

```diff
diff --git a/tools/c2chapel/c2chapel.py b/tools/c2chapel/c2chapel.py
index af758b346f..1e9c3540eb 100755
--- a/tools/c2chapel/c2chapel.py
+++ b/tools/c2chapel/c2chapel.py
@@ -574,7 +574,11 @@ def findIgnores():
                     if line.startswith("typedef"):
                         rhs = line.replace(";", "")
                         rhs = rhs.replace("typedef int ", "")
+                        rhs = rhs.replace("typedef uint32_t ", "")
                         rhs = rhs.replace("typedef _Bool ", "")
+                        rhs = rhs.replace("typedef void* ", "")
+                        if "typedef struct" in rhs:
+                            rhs = rhs.split()[-1]
 
                         ret.add(rhs.strip())
 
```

Which adjusts `findIgnores` to account for things like:

```c
typedef uint32_t xcb_window_t;
typedef void* MirEGLNativeWindowType;
typedef struct MirDisplayConfig MirDisplayConfig;
```

I'm going to run with this and see if we can make `findIgnores` a little more general.

**Comment by ronawho** [comment]
Eh, for now I'm just going to hardcode to keep things simpler for 1.24.1. I'll look at making it more general after that. I've opened https://github.com/chapel-lang/chapel/pull/17519 to do this. I don't care if we merge that, or cherry-pick those changes into this PR.

**Comment by ronawho** [comment]
Chatting with @soohoonchoi we're closing this in favor of https://github.com/chapel-lang/chapel/pull/17519 and https://github.com/chapel-lang/chapel/pull/17520

**Pull request closed without merging**